Source/WebCore/editing/cocoa/HTMLConverter.mm

```diff
@@ -2399,8 +2399,11 @@ AttributedString editingAttributedString(const SimpleRange& range, IncludeImages
     for (TextIterator it(range); !it.atEnd(); it.advance()) {
         auto node = it.node();
 
-        if (RefPtr imageElement = dynamicDowncast<HTMLImageElement>(node); imageElement && includeImages == IncludeImages::Yes)
-            [string appendAttributedString:attributedStringWithAttachmentForElement(*imageElement).get()];
+        if (RefPtr imageElement = dynamicDowncast<HTMLImageElement>(node); imageElement && includeImages == IncludeImages::Yes) {
+            RetainPtr attachmentAttributedString = attributedStringWithAttachmentForElement(*imageElement);
+            [string appendAttributedString:attachmentAttributedString.get()];
+            stringLength += [attachmentAttributedString length];
```

Comment:
Is this compatible with:

`auto currentTextLength = it.text().length();`

and

`[string setAttributes:attrs.get() range:NSMakeRange(stringLength, currentTextLength)];` below,

given that `TextIterator it(range)` does not have any additional behaviors specified?

Wouldn't it result in length mismatches?

Reply:
it _wasn't_ compatible before, but is now. This change doesn't _actually_ affect the string lengths, just changes the position of where the text that comes after the attachment goes.

Reply:
(My tests also exercise this thoroughly)